content/developers/internals/data-model/usage-example.en.md

```diff
@@ -0,0 +1,275 @@
+
+Therefore, if you want to place an object at the beginning of the track section, you would set its offset to 0. To place it at the end of the track section, its offset should be the `length` of the track section.
+{{% /alert %}}
+
+For the objects to be complete, there are different attributes to set:
+
+* `length`: the length of the track section in meters.
+* `geo`: the coordinates in real life (geo is for geographic), in the [GeoJSON](https://en.wikipedia.org/wiki/GeoJSON) format.
+* `sch`: the coordinates in the schematic view (sch for schematic), also in [GeoJSON](https://en.wikipedia.org/wiki/GeoJSON) format.
+* cosmetic attributes: `line_name`, `track_name`, `track_number` which are used to indicate the name and labels that were given to the tracks / lines in real life.
+
+For all track sections in our infrastructure, the `geo` and `sch` attributes are identical, and very much resemble the given schema.
+
+For most track sections, their `length` is proportional to what can be seen in the schema, but for readability, we made *TA6*, *TA7*, *TD0* and *TD1* not correspond to their length (which are 10km and 25km).
+
+### Track Section Links
+
+For the moment we only created track sections, they have length and coordinates but they are not connected to each other (we do not check which coordinates overlap, thank you very much).
+
+If you only need to connect two `TackSections` together, this is where `TrackSectionLinks` come into play. A train can go from one track section to the other only if they are connected by a `TrackSectionLink` (or by a `Switch`).
+
+If you want need to connect more `TackSections` together, you will need the [`Switches`](#switches) objects.
+
+In our infrastructure, since we chose to have our track sections as long as possible, we do not actually need to use `TrackSectionLinks`.
+
+### Switches
+
+A `Switch` object can be understood as a manageable superposition of `TrackSectionLinks`: when instantiating the `Switch`, you define all the connections that you want, and in the *OSRD* core, all these connections will be taken into account, however only one can be active at the same time. Indeed, since we are talking about some sort of overlap of rail, if two connections are activated at once, then two trains could want to use them at the same time, and that would result in a collision (not head on, but side to side, in french called *en écharpe*).
+
+In general, in an infrastructure, switches more or less use the same connection layouts, so in accordance with [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), we have `SwitchType` objects, which gather what connections should be created, when creating this particular type of switch. We describe this object in the next subsection.
```

Comment:
```suggestion
In an infrastructure, switches more or less use the same connection layouts so, to avoid repetitions, we have `SwitchType` objects, which gather what connections should be created, when creating this particular type of switch. We describe this object in the next subsection.
```